src/box/lua/config/utils/log.lua

```diff
@@ -0,0 +1,135 @@
+-- Logger wrapper with a few enhancements.
+--
+-- 1. Encode tables into JSON.
+-- 2. Enable all the messages on TT_CONFIG_DEBUG=1.
+--
+-- Hopefully, the wrapper can be dropped when the following logger
+-- problems will be solved.
+--
+-- * JSONify tables at logging (gh-8611)
+-- * Follow TT_LOG_LEVEL before log.cfg()/box.cfg() (gh-8092)
+--
+-- A solution of the following problems may simplify the code of
+-- this wrapper.
+--
+-- * mylog.cfg() configures the main logger (gh-8610)
+-- * No easy way to determine if a message should go to the log
+--   (gh-8730)
+
+local json_noexc = require('json').new()
+json_noexc.cfg({encode_use_tostring = true})
+
+local logger_name = 'tarantool.config'
+local log = require('log').new(logger_name)
+
+local func2level = {
+    [log.error] = 2,
+    [log.warn] = 3,
+    [log.info] = 5,
+    [log.verbose] = 6,
+    [log.debug] = 7,
+}
+
+local func2prefix = {
+    [log.error] = 'E> ',
+    [log.warn] = 'W> ',
+    [log.info] = 'I> ',
+    [log.verbose] = 'V> ',
+    [log.debug] = 'D> ',
+}
+
+local str2level = {
+    ['fatal'] = 0,
+    ['syserror'] = 1,
+    ['error'] = 2,
+    ['warn'] = 3,
+    ['crit'] = 4,
+    ['info'] = 5,
+    ['verbose'] = 6,
+    ['debug'] = 7,
+}
+
+local function say_closure(log_f)
+    local prefix = ''
+
+    -- Enable logging of everything if an environment variable is
+    -- set.
+    --
+    -- Useful for debugging.
+    --
+    -- Just setting of...
+    --
+    -- ```
+    -- TT_LOG_MODULES='{"tarantool.config": "debug"}'
+    -- ```
+    --
+    -- ...is not suitable due to gh-8092: messages before first
+    -- box.cfg() are not shown.
+    --
+    -- Explicit calling of...
+    --
+    -- ```
+    -- log.cfg({modules = {['tarantool.config'] = 'debug'}})
+    -- ```
+    --
+    -- is not suitable as well, because it makes the logger
+    -- already configured and non-dynamic options like
+    -- `box_cfg.log_nonblock` can't be applied at first box.cfg()
+    -- invocation.
+    --
+    -- So just prefix our log messages and use log.info().
+    local envvar = os.getenv('TT_CONFIG_DEBUG')
+    if envvar ~= nil and envvar ~= '0' and envvar ~= 'false' then
+        prefix = func2prefix[log_f]
+        log_f = log.info
```

Comment:
Otherwise I would always write `I>`. The prefix is to preserve information about the original message's log level. Deserves a short explanation in the code?

Reply:
I have added the following comment:

```lua
-- The prefix represents the original log level of the
-- message.
```

I hope it makes the logic more clear.

Reply:
> Otherwise I would always write `I>`.

Ah, right. Thanks for adding a comment.